packages/react/src/components/UIShell/SideNav.js

```diff
@@ -40,6 +40,16 @@ const SideNav = React.forwardRef(function SideNav(props, ref) {
   const [expandedViaHoverState, setExpandedViaHoverState] =
     useState(defaultExpanded);
   const expanded = controlled ? expandedProp : expandedState;
+
+  // Closing the SideNav when the 'ESC' key is pressed
+  document.addEventListener('keydown', function (event) {
```

Comment:
same comment here as above about `useEvent` and attaching the listener in the DOM